OvernightPrintsSpecBindings/Utils/Utils.cs

```diff
@@ -62,15 +77,17 @@ private static string Rnd(string email, string placeholder)
 				}
 			}
 			catch (FormatException e)
-			{
-				return String.Empty;
-			}
+			{ }
+
+			string leftSide = placeholder.Substring(0, placeholder.IndexOf(temp));
+			string rightSide = placeholder.Substring(placeholder.IndexOf(temp) + temp.Length);
 
-			return left + result + right;
+			return leftSide + result + rightSide;
 		}
 
-		private static string Context(string key)
+		private static string context(string placeholder)
 		{
+			string key = Regex.Match(placeholder, regexRight).ToString();

```

Comment:
Don't parse string here. This method should only take key and return context result